File: cc/math/WaterBottles.cc
/**
 * LeetCode 1518. Water Bottles
 * https://leetcode.com/problems/water-bottles/
 */
class WaterBottles {
public:
    int numWaterBottles(int numBottles, int numExchange) {
        int bottlesSum = numBottles;
        int remainder = 0;
        while (numBottles >= numExchange) {
            remainder = numBottles % numExchange;
            numBottles /= numExchange;
            bottlesSum += numBottles;
            numBottles += remainder;
        }
        return bottlesSum;
    }
};

int main(int argc, char const *argv[]) {
    // ...
    return 0;
}

/* EOF */

File: typescript/math/WaterBottles.ts
/**
 * LeetCode 1518. Water Bottles
 * https://leetcode.com/problems/water-bottles/
 */
function numWaterBottles(numBottles: number, numExchange: number): number {
    let sumBottles: number = numBottles;
    let remainder: number = 0;
    while (numBottles >= numExchange) {
        remainder = numBottles % numExchange;
        numBottles = Math.floor(numBottles / numExchange);
        sumBottles += numBottles;
        numBottles += remainder;
    }
    return sumBottles;
};

// EOF
